lang/src/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -110,25 +110,25 @@ val ExprValue.address: ExprValue?
     get() = asFacet(Addressed::class.java)?.address
 
 fun ExprValue.booleanValue(): Boolean =
-    scalar.booleanValue() ?: errNoContext("Expected boolean: $ionValue", errorCode = ErrorCode.EVALUATOR_UNEXPECTED_VALUE_TYPE, internal = false)
+    scalar.booleanValue() ?: errNoContext("A value of BOOLEAN type is expected, however, what's passed is: $ionValue, with type of ${ionValue.type}", errorCode = ErrorCode.EVALUATOR_UNEXPECTED_VALUE_TYPE, internal = false)
```

Comment:
The error message is more descriptive, but it's not quite what I had in mind for the previous [comment](https://github.com/partiql/partiql-lang-kotlin/pull/604/files#r870764266). For the aggregation functions that only accept numeric types (e.g. `SUM` and `AVG`), I felt the error could make it more explicit that a non-numeric value was provided to the aggregation function. I felt the current error on `ExprValue` doesn't make it clear about the numeric requirement for those aggregation functions. Let me know if this distinction is unclear.

A better error message imo would be something like:
```
org.partiql.lang.eval.EvaluationException: SUM expects number values but was provided "abc" with type of STRING.
```